realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/EncryptedSynchronizedRealmTests.java

```diff
@@ -0,0 +1,254 @@
+package io.realm.objectserver;
+
+import android.os.SystemClock;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import io.realm.ObjectServerError;
+import io.realm.Realm;
+import io.realm.RealmResults;
+import io.realm.SyncConfiguration;
+import io.realm.SyncCredentials;
+import io.realm.SyncManager;
+import io.realm.SyncSession;
+import io.realm.SyncUser;
+import io.realm.TestHelper;
+import io.realm.entities.StringOnly;
+import io.realm.exceptions.RealmFileException;
+import io.realm.objectserver.utils.Constants;
+import io.realm.objectserver.utils.StringOnlyModule;
+import io.realm.objectserver.utils.UserFactory;
+import io.realm.rule.TestSyncConfigurationFactory;
+import io.realm.util.SyncTestUtils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class EncryptedSynchronizedRealmTests extends BaseIntegrationTest {
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(10);
+
+    @Rule
+    public final TestSyncConfigurationFactory configurationFactory = new TestSyncConfigurationFactory();
+
+    @Before
+    public void before() {
+        // This will set the 'm_metadata_manager' in 'sync_manager.cpp' to be 'null'
+        // causing the SyncUser to remain in memory.
+        // They're actually not persisted into disk.
+        // move this call to `tearDown` to clean in-memory & on-disk users
+        // once https://github.com/realm/realm-object-store/issues/207 is resolved
+        SyncTestUtils.resetSyncMetadata();
+    }
+
+    // Make sure the encryption is local, i.e after deleting a synced Realm
+    // re-open it again with no (or different) key, should be possible.
+    @Test
+    public void setEncryptionKey_canReOpenRealmWithoutKey() {
+
+        // STEP 1: open a synced Realm using a local encryption key
+        String username = UUID.randomUUID().toString();
+        String password = "password";
+        SyncUser user = SyncUser.login(SyncCredentials.usernamePassword(username, password, true), Constants.AUTH_URL);
+
+        final byte[] randomKey = TestHelper.getRandomKey();
+
+        SyncConfiguration configWithEncryption = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM)
+                .modules(new StringOnlyModule())
```

Comment:
It should be there, but it is package protected so only works if the test is in `io.realm`